Okta/OktaTokenManager.swift

```diff
@@ -89,33 +73,31 @@ open class OktaTokenManager: NSObject, NSCoding {
         try? self.init(
                     authState: decoder.decodeObject(forKey: "authState") as! OIDAuthState,
                        config: decoder.decodeObject(forKey: "config") as! [String: String],
-                accessibility: (decoder.decodeObject(forKey: "accessibility") as! CFString),
-            validationOptions: (decoder.decodeObject(forKey: "validationOptions") as! [String: Any])
+                accessibility: (decoder.decodeObject(forKey: "accessibility") as! CFString)
         )
     }
 
     public func encode(with coder: NSCoder) {
         coder.encode(self.authState, forKey: "authState")
         coder.encode(self.config, forKey: "config")
         coder.encode(self.accessibility, forKey: "accessibility")
-        coder.encode(self.validationOptions, forKey: "validationOptions")
     }
 
     public func isValidToken(idToken: String?) throws -> Bool {
-        guard let token = idToken else { return false }
-        do {
-            let isValid = try Introspect().validate(jwt: token, options: self.validationOptions)
-            if isValid {
-                return true
-            }
-        } catch let error {
-            throw error
+        guard let idToken = idToken,
+            let tokenObject = OIDIDToken(idTokenString: idToken) else {
+                throw OktaError.JWTDecodeError
+        }
+        
+        if tokenObject.expiresAt.timeIntervalSinceNow < 0 {
```

Comment:
Can this be removed? This appears to already be captured within `OIDIDToken` and validation is [performed](https://github.com/openid/AppAuth-iOS/blob/2a7a6754fd5b612aa554fa56d6e7853218e2d84f/Source/OIDAuthorizationService.m#L455-L465)

Reply:
Yes, but token can expire **after** it passed `AppAuth` validation. For example if you checking it after few days.

Reply:
@nbarbettini & @robertjd - When we support RP-Initiated logout, this is going to cause problems.

The `/logout` endpoint can be passed tokens that are **expired**, and if we continue to throw an error when the token expires, it becomes impossible to terminate the Okta session.

I propose we return expired tokens (remove this conditional and the check [here](https://github.com/okta/okta-sdk-appauth-ios/blob/cf09b2e25c581dad8f64824213a4efbb4d98033b/Okta/OktaTokenManager.swift#L27)), but add the `exp` logic to our [`isAuthenticated` method](https://github.com/okta/okta-sdk-appauth-ios/blob/cf09b2e25c581dad8f64824213a4efbb4d98033b/Okta/OktaAppAuth.swift#L41-L56).